src/main/java/io/cdap/e2e/pages/actions/CdfConnectionActions.java

```diff
@@ -0,0 +1,217 @@
+   *
+   * @param connectionName If connectionName is present in {@link ConstantsUtil#DEFAULT_PLUGIN_PROPERTIES_FILE} as key
+   *                       then actual connectionName is fetched from it
+   *                       else connectionName param used as it is.
+   */
+  public static void verifyConnectionIsCreatedSuccessfully(String connectionName) {
+    String connectionNameFromPluginPropertiesFile = PluginPropertyUtils.pluginProp(connectionName);
+    if (connectionNameFromPluginPropertiesFile == null) {
+      WaitHelper.waitForElementToBeDisplayed(CdfPluginPropertiesLocators.locateElementWithText(connectionName));
+      return;
+    }
+    WaitHelper.waitForElementToBeDisplayed(CdfPluginPropertiesLocators
+                                             .locateElementWithText(connectionNameFromPluginPropertiesFile));
+  }
+
+  /**
+   * Select data row for connection
+   *
+   * Based on the connection type - table with rows containing data (i.e.bucket, dataset, table, directory names etc.)
+   * will be displayed.
+   * To select data for the connection use this method with its name as parameter.
+   * Example : For GCS pass actual bucket name as parameter.
+   */
+  private static void selectConnectionDataRow(String dataRow) {
+    WaitHelper.waitForElementToBeHidden(CdfConnectionLocators.locatorOfConnectionDataLoadingIndicator,
+                                        ConstantsUtil.DEFAULT_TIMEOUT_SECONDS);
+    ElementHelper.sendKeys(CdfConnectionLocators.searchDirectoryInput, dataRow);
+    int attempts = 0;
+    while (attempts < 5) {
+      try {
+        ElementHelper.hoverOverElement(CdfPluginPropertiesLocators.locateElementWithText(dataRow));
+        ElementHelper.clickOnElement(CdfPluginPropertiesLocators.locateElementWithText(dataRow));
+        break;
+      } catch (StaleElementReferenceException e) {
+        if (attempts == 4) {
+          throw e;
+        }
+      }
+      attempts++;
+    }
+  }
+
+  /**
+   * Select data for connection using name
+   *
+   * Example :
+   * For GCS, table with bucket names will be displayed.
+   * To select bucket for the connection use this method with bucket name as parameter.
+   *
+   * Once bucket name is selected - in next table all directories inside bucket will be displayed.
+   * To select directory use this method with directory name as parameter.
+   *
+   *
+   * @param dataName If dataName is present in {@link ConstantsUtil#DEFAULT_PLUGIN_PROPERTIES_FILE} as a key
+   *                 then actual dataName to select is fetched from it
+   *                 else dataName used as it is.
+   */
+  public static void selectConnectionDataWithName(String dataName) {
+    String dataToSelect = PluginPropertyUtils.pluginProp(dataName);
+    if (dataToSelect == null) {
+      dataToSelect = dataName;
+    }
+    selectConnectionDataRow(dataToSelect);
+  }
+
+  /**
+   * Select data rows for connection traversing through directories using provided path
```

Comment:
Please add an example `path` in comments.

Reply:
Added example path in comment.